### PR TITLE
Characters escaping

### LINKED ISSUE #13: Character escaping issue
It would be wonderful if the solution automatically escaped strings used as labels and the title. I mean quote characters, line breaks, and others that I forgot to mention.

What I noticed so far is:
- diagram title is not quoted automatically when it contains spaces, and this breaks the structure of the compiled diagram,
- special characters in labels, and probably in the title as well, are not escaped (I did not check quoting when required).

I didn't familiarize with the DOT syntax enough, but from my experience with the library:
- line breaks have to be manually replaced with `\n` (some suggest using `<br />`, but it didn't work for me),
- quote characters have to be escaped with a backslash `"` -> `\"`,
- same case with backslashes: `\` -> `\\`
- I also escaped labels and title with `HttpUtility.HtmlEncode(label)` to handle special HTML characters like `<`, `>`, national characters (they also caused issues), etc.

Only then did the online visualizer http://www.webgraphviz.com/ stop reporting errors and warnings.

By the way, I love your library, it is so helpful! Cheers!

### PR DESCRIPTION
**Pull request by vfrz**
Fix #13

### CODE REVIEW

File: Sources/DotNetGraph/Compiler/DotCompiler.cs [hunk 1]
```diff
@@ -224,7 +222,9 @@ private void CompileAttributes(StringBuilder builder, ReadOnlyCollection<IDotAtt
 
         private static string FormatString(string value)
         {
-            var result = value.Replace("\"", "\\\"");
+            var result = value
+                .Replace("\\", "\\\\")
+                .Replace("\"", "\\\"");
```
Comment (mariusz-schimke) [inline Sources/DotNetGraph/Compiler/DotCompiler.cs]: To answer your [question](https://github.com/vfrz/DotNetGraph/issues/13#issuecomment-619546581), I was thinking about this: `.Replace(Environment.NewLine, @"\n")`.

Notice, however, that `\n` is not a C# line break char - it is a string literal consisting of two characters: a backslash and n.

Comment (vfrz) [reply Sources/DotNetGraph/Compiler/DotCompiler.cs]: Ok so the goal here is to remove any platform/environment-specific line break and use `\n` so it's consistent?

Comment (mariusz-schimke) [reply Sources/DotNetGraph/Compiler/DotCompiler.cs]: Let me clarify. In C# '\n' is a single character (LF), right? So this is not about replacing CRLF ("\r\n") with LF. It is about replacing CRLF with two following characters: @"\n" (a backslash and an n letter); `@"\n"` == `"\\n"`.

So this text:

```
First line
Second line
```

Should appear in the DOT label as:

`First line\nSecond line`

Comment (mariusz-schimke) [reply Sources/DotNetGraph/Compiler/DotCompiler.cs]: You should see what I mean when you paste the following graph [here](http://www.webgraphviz.com/):

```
graph G {
    my_node [label="First line\nSecondLine"];
}
```

Comment (vfrz) [reply Sources/DotNetGraph/Compiler/DotCompiler.cs]: Ok, I got it now, I'll make the changes later that day.

Comment (vfrz) [reply Sources/DotNetGraph/Compiler/DotCompiler.cs]: I've made the changes, I've replaced both \r\n and \n to `\\n` so it should works on linux and windows.

Comment (mariusz-schimke) [reply Sources/DotNetGraph/Compiler/DotCompiler.cs]: Perfect, thanks! :)

**Pull request merged**